### PR TITLE
Moving From Utmregistry to this repo and respective changes

### PR DESCRIPTION
**Pull request by Anirudh-zemoso**

### CODE REVIEW

**Comment by Anirudh-zemoso** [comment]
@all tagging stakeholders , to inform about the repo and artifacts.

**Comment by Anirudh-zemoso** [comment]
> Devops and jfrog pipeline ?

Jenkinsfile is created , jfrog pipeline from jenkins job, which @suraj-zemoso is going to take up.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+@Library('airxos-jenkins-pipeline-library@2.0.0') _
+airxosJavaGenericPipeline apiDirectories: ['./ticket-service','./websocket-util'],
+  githubRepoSlug: 'gebeyond/websocket-utils',
+  skipSonarqube: true
```
Comment (rajavikram) [inline Jenkinsfile:4]: @Anirudh-zemoso, Consider enabling sonarqube.

File: ticket-service/src/main/java/io/airxos/ticket/ITicketRequestApprover.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import io.airxos.ticket.model.TicketVenue;
+
+/**
+ * Interface for a function object that is used to determine whether a ticket request is allowed.
+ */
+@FunctionalInterface
+public interface ITicketRequestApprover {
+
+  boolean test(String registryUserId, TicketVenue ticketVenue, String resource);
```
Comment (rajavikram) [inline ticket-service/src/main/java/io/airxos/ticket/ITicketRequestApprover.java:11]: The naming of this method could be improved.

File: ticket-service/src/main/java/io/airxos/ticket/ITicketService.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+   * @param duration       How long the ticket will be valid before expiration.
+   * @param orgId          Organization of the user.
+   * @param teamIds        Teams that the user belongs to.
+   * @return single-use Ticket
+   */
+  Ticket createTicket(String registryUserId,
+                      TicketVenue ticketVenue,
+                      String resource,
+                      Duration duration,
+                      Optional<String> orgId,
```
Comment (rajavikram) [inline ticket-service/src/main/java/io/airxos/ticket/ITicketService.java:29]: Do we ever want to allow a null value for orgId ?

File: ticket-service/src/main/java/io/airxos/ticket/detail/TicketService.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    fEndpoints.put(endpoint, approver);
+  }
+
+  @Override
+  public Ticket createTicket(final String registryUserId,
+                             final TicketVenue ticketVenue,
+                             final String resource,
+                             final Duration duration,
+                             final Optional<String> orgId,
+                             final Set<String> teamIds) {
```
Comment (rajavikram) [inline ticket-service/src/main/java/io/airxos/ticket/detail/TicketService.java:71]: Teams can change while the user is mid-session, we need let the client know to get a new ticket in that case.
Also do we even need the team ids?
FYI @ray-chow-ge, @jqk1032, @Pranava29

File: ticket-service/src/main/java/io/airxos/ticket/model/AbstractTicket.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  abstract Optional<String> getOrgId();
+
+  /**
+   * Get the ids of the user's Teams.
+   */
+  abstract Set<String> getTeamIds();
+
+
+  @Override
+  public String toString() {
```
Comment (rajavikram) [inline ticket-service/src/main/java/io/airxos/ticket/model/AbstractTicket.java:62]: Immutables provided toString should suffice for this use case.

File: ticket-service/src/main/java/io/airxos/ticket/model/ModelConstants.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
+package io.airxos.ticket.model;
+
+import io.airxos.core.util.Preconditions;
+import java.util.Arrays;
+import javax.annotation.Nullable;
+
+public class ModelConstants {
```
Comment (rajavikram) [inline ticket-service/src/main/java/io/airxos/ticket/model/ModelConstants.java:7]: We will have to revisit this class name and the values it's containing.

**Pull request merged**